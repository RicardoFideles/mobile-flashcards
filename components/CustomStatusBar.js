import React from 'react';
import { View, StatusBar } from 'react-native';
import { Constants } from 'expo';
import PropTypes from 'prop-types';

const CustomStatusBar = ({ backgroundColor, ...props }) => (
  <View style={{ backgroundColor, height: Constants.statusBarHeight }}>
    <StatusBar translucent backgroundColor={backgroundColor} {...props} />
  </View>
);

CustomStatusBar.propTypes = {
  backgroundColor: PropTypes.string.isRequired,
};

export default CustomStatusBar;
