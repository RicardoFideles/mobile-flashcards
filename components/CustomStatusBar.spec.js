import React from 'react';
import renderer from 'react-test-renderer';
import CustomStatusBar from './CustomStatusBar';

describe('CustomStatusBar', () => {
  it('renders correctly', () => {
    const wrappper = renderer
      .create(<CustomStatusBar backgroundColor="red" />)
      .toJSON();
    expect(wrappper).toMatchSnapshot();
  });
});
