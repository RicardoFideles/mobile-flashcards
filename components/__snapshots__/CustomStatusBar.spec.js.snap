// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomStatusBar renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "red",
      "height": 44,
    }
  }
/>
`;
